askup/utils/views.py

```diff
@@ -466,3 +466,83 @@ def pop_next_existent_question_id_from_list(question_ids):
 
         if Question.objects.filter(id=next_question_id).exists():
             return next_question_id
+
+
+def get_redirect_on_answer_fail(request, qset_id, filter, is_quiz):
+    """
+    Redirect to a backup question if answering question is failing due to deletion of question.
+
+    Returns HttpResponseRedirect to the qset where question was located, to the next question
+    in the quiz or to the organizations list.
+    """
+    if is_quiz:
+        return get_json_redirect_next_quiz_question(request.user.id, qset_id, filter)
+
+    return get_json_redirect_qset(qset_id)
+
+
+def get_json_redirect_organizations_list():
+    """
+    Return the json object with redirect to the organizations list.
+    """
+    notification = (
+        'This question was deleted since you\'ve opened it,' +
+        'redirecting you to your organizations list ...'
+    )
+    return JsonResponse(
+        {
+            'result': 'error',
+            'redirect_url': reverse('askup:organizations'),
+            'notification': notification,
+        }
+    )
+
+
+def get_json_redirect_next_quiz_question(user_id, qset_id, filter):
+    """
+    Return the json object with redirect to the next question in the quiz.
+    """
+    next_question_id = get_next_quiz_question(user_id, filter, qset_id, False)
+    notification = (
+        'This question was deleted since you\'ve opened it,' +
+        'redirecting you to the next one in the Quiz...'
+    )
+    redirect_url = reverse(
+        'askup:question_answer',
+        kwargs={
+            'question_id': next_question_id,
+            'qset_id': qset_id,
+        }
+    )
+    return JsonResponse(
+        {
+            'result': 'error',
+            'redirect_url': redirect_url,
+            'notification': notification,
+        }
+    )
+
+
+def get_json_redirect_qset(qset_id):
+    """
+    Return the json object with redirect to the qset.
+    """
+    notification = (
```

Comment:
This notification text is almost the same in all function why don't move it to the module's variable?

Reply:
done

Reply:
`notification = QUESTION_DELETED_TEXT.format('the correspondent subject...')` is fair enough multistring split is not needed here

Reply:
missed those brackets